src/io/flutter/devtools/DevToolsManager.java

```diff
@@ -187,58 +187,62 @@ private boolean isBazel(Project project) {
   }
 
   public void openBrowserIntoPanel(FlutterApp app, String uri, ContentManager contentManager, String tabName, String pageName) {
-    final String screen = null;
+    ApplicationManager.getApplication().executeOnPooledThread(() -> {
+      final String screen = null;
 
-    if (isBazel(project)) {
-      try {
-        getDevToolsInstance(app).get(15, TimeUnit.SECONDS).openPanel(project, uri, contentManager, tabName, pageName);
-      }
-      catch (Exception e) {
-        LOG.info("Failed to get existing devToolsInstance");
-      }
-    }
-    else {
-      if (devToolsInstance != null) {
-        devToolsInstance.openPanel(project, uri, contentManager, tabName, pageName);
+      if (isBazel(project)) {
+        try {
+          getDevToolsInstance(app).get(15, TimeUnit.SECONDS).openPanel(project, uri, contentManager, tabName, pageName);
+        }
+        catch (Exception e) {
+          LOG.info("Failed to get existing devToolsInstance");
+        }
       }
       else {
-        @Nullable final OSProcessHandler handler = getProcessHandlerForPub();
-
-        if (handler != null) {
-          // start the server
-          DevToolsInstance.startServer(handler, instance -> {
-            devToolsInstance = instance;
-
-            devToolsInstance.openPanel(project, uri, contentManager, tabName, pageName);
-          }, () -> {
-            // Listen for closing; null out the devToolsInstance.
-            devToolsInstance = null;
-          }, project);
+        if (devToolsInstance != null) {
+          devToolsInstance.openPanel(project, uri, contentManager, tabName, pageName);
+        }
+        else {
+          @Nullable final OSProcessHandler handler = getProcessHandlerForPub();
+
+          if (handler != null) {
+            // start the server
+            DevToolsInstance.startServer(handler, instance -> {
+              devToolsInstance = instance;
+
+              devToolsInstance.openPanel(project, uri, contentManager, tabName, pageName);
```

Comment:
another time gap. Make sure the project is still valid before we open the panel.